src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,218 @@
+      if (!await this.sendData()) {
+        this.logger.error('Impossible to send data')
+        return ApiHandler.STATUS.LOGIC_ERROR
+      }
+    }
+
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - Path of the file to parse
+   * @return {Promise} - the parsed content of the file
+   */
+  async parseCsvFile(filePath) {
+    return new Promise((resolve, reject) => {
+      // Initialize array wich will be filled at each step
+      const csvDataParsed = []
+      const csvFile = fs.createReadStream(filePath)
+
+      // Parse the csv file into a json object
+      Papa.parse(csvFile, {
+        delimiter: this.csvDelimiter,
+        header: true,
+        dynamicTyping: true,
+        error: (error) => {
+          this.logger.error(error)
+          reject()
+          return []
+        },
+        step: (step) => {
+          csvDataParsed.push(step.data)
+        },
+        complete: () => {
+          resolve(csvDataParsed)
+        },
+      })
+    })
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {void}
+   */
+  convertToHttpBody(jsonObject) {
+    // Log all headers in the csv file and log the value ine the mapping not present in headers
+    this.logger.silly(`All available headers are: ${Object.keys(jsonObject[0])}`)
+
+    this.mapping.forEach((mapping) => {
+      if (!Object.prototype.hasOwnProperty.call(jsonObject[0], mapping.csvField)) {
+        this.logger.debug(`The header: '${mapping.csvField}' is not present in the csv file`)
+      }
+    })
+
+    // Start the mapping for each row
+    jsonObject.forEach((jsonObj) => {
+      const response = this.convertCSVRowIntoHttpBody(jsonObj)
+
+      // Test the result of the mapping/convertion before add it in the httpBody
+      if (response) {
+        this.httpBody.push(response)
+      }
+    })
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Object} - Object mapped for one row
+   */
+  convertCSVRowIntoHttpBody(jsonObject) {
+    const object = {}
+
+    this.mapping.forEach((mapping) => {
+      if (Object.prototype.hasOwnProperty.call(jsonObject, mapping.csvField)) {
+        const field = mapping.httpField
+        const value = CsvToHttp.convertToCorrectType(jsonObject[mapping.csvField], mapping.type)
+
+        object[field] = value
+      }
+    })
+    return object
+  }
+
+  /**
+   * Convert the value in the selected type (string by default)
+   * @param {Mixed} value - value
+   * @param {string} type - type
+   * @return {Mixed} - The converted value
+   */
+  static convertToCorrectType(value, type) {
+    let valueConverted
+    switch (type) {
+      case 'integer': return parseInt(value, 10) ? parseInt(value, 10) : value
+      case 'float': return parseFloat(value) ? parseFloat(value) : value
+      case 'timestamp': return parseInt(value, 10) ? parseInt(value, 10) : value
+      case 'date (ISO)': {
+        valueConverted = new Date(value)
```

Comment:
this one is not acceptable. a value global to the scope and reused in several places is a good recipe for bugs. 
please repeat the declaration for each scope